pkg/importer/openapi.go

```diff
@@ -3,105 +3,150 @@ package importer
 import (
 	"bytes"
 	"fmt"
-	"io/ioutil"
 	"net/url"
 	"path/filepath"
+	"runtime"
 	"sort"
 	"strings"
 
 	"github.com/getkin/kin-openapi/openapi3"
 	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
 )
 
-const openapiv3DefinitionPrefix = "#/components/schemas/"
+type openapiv3 struct {
+	appName  string
+	pkg      string
+	basePath string // Used for Swagger2.0 files which have a basePath field
+	logger   *logrus.Logger
+	fs       afero.Fs
+	types    TypeList
 
-func MakeOpenAPI3Importer(logger *logrus.Logger, basePath string, filePath string) *OpenAPI3Importer {
-	return &OpenAPI3Importer{
-		logger:            logger,
-		externalSpecs:     make(map[string]*OpenAPI3Importer),
-		types:             TypeList{},
-		intermediateTypes: TypeList{},
-		basePath:          basePath,
-		swaggerRoot:       filePath,
-	}
+	nameStack []string
 }
 
-type OpenAPI3Importer struct {
-	appName       string
-	pkg           string
-	basePath      string // Used for Swagger2.0 files which have a basePath field
-	logger        *logrus.Logger
-	externalSpecs map[string]*OpenAPI3Importer
-	spec          *openapi3.Swagger
-	types         TypeList
-	// intermediateTypes is a temporary list which places the type is in parsing process still.
-	// It can help to support circular dependency, like type A has an array contains type A itself.
-	intermediateTypes TypeList
-	swaggerRoot       string
-	globalParams      Parameters
-}
-
-func (l *OpenAPI3Importer) Load(input string) (string, error) {
+func NewOpenAPILoader(logger *logrus.Logger, fs afero.Fs) *openapi3.SwaggerLoader {
```

Comment:
nit: it seems NewOpenAPILoader isn't used outside the `import` package, should it be public or should it be made private to encourage the use of the `NewOpenAPIV3Importer` API